src/modules/gnd_pos_control/GroundRoverPositionControl.cpp

```diff
@@ -257,51 +262,33 @@ GroundRoverPositionControl::control_position(const matrix::Vector2f &current_pos
 			}
 		}
 
-		if (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_IDLE) {
-			_att_sp.roll_body = 0.0f;
-			_att_sp.pitch_body = 0.0f;
-			_att_sp.yaw_body = 0.0f;
-			_att_sp.thrust_body[0] = 0.0f;
+		if (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_IDLE ||
+		    pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LOITER ||
+		    pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LAND) {
+
+			_act_controls.control[actuator_controls_s::INDEX_YAW] = 0.0f;
+			_act_controls.control[actuator_controls_s::INDEX_THROTTLE] = 0.0f;
 
 		} else if ((pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_POSITION)
 			   || (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF)) {
 
 			/* waypoint is a plain navigation waypoint or the takeoff waypoint, does not matter */
 			_gnd_control.navigate_waypoints(prev_wp, curr_wp, current_position, ground_speed_2d);
-			_att_sp.roll_body = _gnd_control.get_roll_setpoint();
-			_att_sp.pitch_body = 0.0f;
-			_att_sp.yaw_body = _gnd_control.nav_bearing();
-			_att_sp.fw_control_yaw = true;
-			_att_sp.thrust_body[0] = mission_throttle;
-
-		} else if (pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LOITER) {
-
-			/* waypoint is a loiter waypoint so we want to stop*/
-			_gnd_control.navigate_loiter(curr_wp, current_position, pos_sp_triplet.current.loiter_radius,
-						     pos_sp_triplet.current.loiter_direction, ground_speed_2d);
-
-			_att_sp.roll_body = _gnd_control.get_roll_setpoint();
-			_att_sp.pitch_body = 0.0f;
-			_att_sp.yaw_body = _gnd_control.nav_bearing();
-			_att_sp.fw_control_yaw = true;
-			_att_sp.thrust_body[0] = 0.0f;
-		}
 
-		if (was_circle_mode && !_gnd_control.circle_mode()) {
-			/* just kicked out of loiter, reset integrals */
-			_att_sp.yaw_reset_integral = true;
+			_act_controls.control[actuator_controls_s::INDEX_THROTTLE] = mission_throttle;
+
+			float desired_r = ground_speed_2d.norm_squared() / math::abs_t(_gnd_control.nav_lateral_acceleration_demand());
+			//desired_r = math::constrain(desired_r, 0.0000001f, 9999999999.0f);
+			float desired_theta = (0.5f * (float) M_PI) - (float) std::atan2(desired_r, _parameters.wheel_base);
```

Comment:
```suggestion
			float desired_theta = (0.5f * M_PI_F) - atan2f(desired_r, _parameters.wheel_base);
```

Long story about the current state of NuttX c++, but better to use atan2f (math.h) directly for now.